frontend/lib/src/WidgetStateManager.ts

```diff
@@ -216,6 +216,10 @@ export class WidgetStateManager {
     const form = this.getOrCreateFormState(formId)
 
     const submitButtons = this.formsData.submitButtons.get(formId)
+    const disableForm = submitButtons?.at(0)?.disabled
```

Comment:
Why the first submit button specifically? We probably don't have many cases where there's >1 submit button in a form to begin with, but I'd guess that in those cases we should just disable the whole form if any of the buttons are disabled

Reply:
That's what my first thought was but I went to talk to johannes and he said to just disable the first as we call submitting the form with the first submit button so technically it's the "same" behavior from an implementation perspective perspective. I don't have a strong opinion here so happy to change.

Cc: @jrieke

Reply:
I took some time to think about it, and I think I would agree with @vdonato on this one. I'll start by saying that there's an edge case here, and I think if there are multiple submit buttons and one is disabled and one is not, it would be reasonable to say that this is a bad invariant, and therefore can have undefined behavior. If I were to implement the best solution it would be:

- disable submission if any submit button is disabled. Forms can have destructive outcomes when submitted, and if there's a bug that caused a destructive experience. I'd rather be conservative here.
- Because this is such a weird case, I think we should identify the bad invariant. I think we can identify the bad variant [here](https://github.com/streamlit/streamlit/blob/develop/frontend/lib/src/components/core/Block/Block.tsx#L96).

Reply:
Reason why we went with that solution is that pressing Enter always submits the first submit button currently. So it kinda makes sense to check if the first submit button is disabled. But thinking about it again now, "we should just disable the whole form if any of the buttons are disabled" actually makes sense to me. So happy to go with that behavior if y'all think it's better! (Sorry I didn't see this before.)